web-interface/api/logic.py

```diff
@@ -101,13 +101,33 @@ def AppendDiskInfo(diskMountPoint):
                  })
     pass
 
-# Going to be updating to increase efficiency of nmap scans.
+
+# Create a list of ports and services, and create a class to define ports and services.
+ServiceList = []
+class PortInfo:
+    def __init__(self, port, service, IsActive):
+        self.port = port
+        self.service = service
+        self.IsActive = IsActive
+# Define ports. IsActive will be "False" by default. Becomes "True" it the port is Open.
+port22 = PortInfo(22, 'ssh', False)
```

Comment:
This is hardcoding the ports as a separate variable. What you can do is to create the object with a function, and append this created object to a list. This is described in the pseudo-code in the issue.